include/sys_clock.h

```diff
@@ -125,23 +125,21 @@ static ALWAYS_INLINE s32_t _ms_to_ticks(s32_t ms)
 #endif
 }
 
-static inline s64_t __ticks_to_ms(s64_t ticks)
+static inline u64_t __ticks_to_ms(s64_t ticks)
 {
 #ifdef CONFIG_SYS_CLOCK_EXISTS
 
 #ifdef _NEED_PRECISE_TICK_MS_CONVERSION
 	/* use 64-bit math to keep precision */
-	return (u64_t)ticks * MSEC_PER_SEC / CONFIG_SYS_CLOCK_TICKS_PER_SEC;
+	return (u64_t)ticks * (MSEC_PER_SEC / (u64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC);
```

Comment:
No, this is a precision bug.  The multiplication has to happen first, because the division will compute something along the lines of 1000/100000000 and produce a zero.

What's the MISRA rule being addressed here?  Just that expressions must be fully parenthesized?  The * and / operators are left-to-right associative and have the same precedence.

Reply:
indeed, left to right evaluation :/
I'll change it.